src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -1066,4 +1079,27 @@ public String getJcrOutputPath() {
     }
   }
 
+  int checkAndGetUserInputQueryLimit( final Object userInputQueryLimit, final int reportQueryLimit ) {
```

Comment:
You are trusting the user's input. Lovely. What happens if I manipulate the URL to get a really large (larger than system limit) user limit into this method?

You'll need a Min(sysLimit, userLimit) check. 
Your unit-test misses the set of tests where the user limit is larger than the system limit while no report-limit is set.

The logic is as follows:

	# If the report has a defined row limit, ignore the user's input, but apply the system limit
	if haveReportLimit
	   if haveSystemLimit
	     return Min(systemLimit, ReportLimit);
	   else
	     return reportLimit
	else
           # system and/or user limit may be undefined. so we need a hell of a if-then-else block ..
	   if haveSystemLimit && haveUserLimit
	     return Min(userLimit, systemLimit);
	   else if haveSystemLimit
	     return systemLimit
	   else if haveUserLimit
	     return userLimit
	   else 
	     return -1